### ISSUE TITLE
PT Rule Request: Don't allow `pytest.raises(AssertionError)`

### ISSUE THREAD

**Issue by notatallshaw** [astral-sh/ruff#12704]
I am working on adding [pytest rules for pip](https://github.com/pypa/pip/pull/12894). And it highlighted [some tests](https://github.com/pypa/pip/blob/3259c061fb36698339877bbf2cefb5e82815e812/tests/functional/test_install_reqs.py#L627) had an interesting anti-pattern:

```python
with pytest.raises(AssertionError):
    assert something in foo
```

Instead of:

```python
assert something in foo
```

The `pytest.raises(AssertionError)` is redundant here, and it creates a confusing test flow to follow, the test condition should simply be reversed.

I assume a similiar issue could happen with a `try/except AssertionError`

**Comment by charliermarsh** [comment]
This seems reasonable to me.

**Comment by charliermarsh** [comment]
@RonnyPfannschmidt may have an opinion here.

**Comment by RonnyPfannschmidt** [comment]
Raises with just assertion error should only be allowed if no assert is in the body 

Its reasonable to also require either a match parameter or follow-up assertions on the Exception info returned 

The transformation of the previously mentioned with block to an negated assertion ought to be a unsafe fix

**Comment by notatallshaw** [comment]
Oh yeah, the example was for demonstration purposes, not as an example to auto fix .

**Comment by RonnyPfannschmidt** [comment]
It's still useful if the language server can propose a fix
